packages/eslint-plugin-pf-codemods/src/rules/v6/cardUpdatedClickableMarkup/card-updated-clickable-markup.test.ts

```diff
@@ -0,0 +1,138 @@
+      errors: [
+        {
+          message: "The markup for clickable-only cards has been updated.",
+          type: "JSXElement",
+        },
+      ],
+    },
+    // 1 valid property + 1 invalid property
+    {
+      code: `import { Card, CardHeader } from '@patternfly/react-core'; <Card isClickable><CardHeader selectableActions={{to: '#', selectableActionId: 'Id'}} /></Card>`,
+      output: `import { Card, CardHeader } from '@patternfly/react-core'; <Card isClickable><CardHeader selectableActions={{to: '#'}} /></Card>`,
+      errors: [
+        {
+          message:
+            "The markup for clickable-only cards has been updated.Additionally, the `selectableActions.selectableActionId` and `selectableActions.name` props are no longer necessary to pass to CardHeader for clickable-only cards.",
+          type: "JSXElement",
+        },
+      ],
+    },
+    // 2 invalid properties
+    {
+      code: `import { Card, CardHeader } from '@patternfly/react-core'; <Card isClickable><CardHeader selectableActions={{name: 'Test', selectableActionId: 'Id'}} /></Card>`,
+      output: `import { Card, CardHeader } from '@patternfly/react-core'; <Card isClickable><CardHeader selectableActions={{}} /></Card>`,
+      errors: [
+        {
+          message:
+            "The markup for clickable-only cards has been updated.Additionally, the `selectableActions.selectableActionId` and `selectableActions.name` props are no longer necessary to pass to CardHeader for clickable-only cards.",
+          type: "JSXElement",
+        },
+      ],
+    },
+    // 1 valid property + 2 invalid properties
+    {
+      code: `import { Card, CardHeader } from '@patternfly/react-core'; <Card isClickable><CardHeader selectableActions={{to: "#", name: 'Test', selectableActionId: 'Id'}} /></Card>`,
+      output: `import { Card, CardHeader } from '@patternfly/react-core'; <Card isClickable><CardHeader selectableActions={{to: "#"}} /></Card>`,
+      errors: [
+        {
+          message:
+            "The markup for clickable-only cards has been updated.Additionally, the `selectableActions.selectableActionId` and `selectableActions.name` props are no longer necessary to pass to CardHeader for clickable-only cards.",
+          // message: `The markup for clickable-only cards has been updated, now using button and anchor elements for the respective clickable action. The \`selectableActions.selectableActionId\` and \`selectableActions.name\` props are also no longer necessary for clickable-only cards. Passing them in will not cause any errors, but running the fix for this rule will remove them.`,
+          type: "JSXElement",
+        },
+      ],
+    },
+    // Passed as a variable reference
+    {
+      code: `import { Card, CardHeader } from '@patternfly/react-core'; const obj = {name: 'Test', selectableActionId: 'Id'}; <Card isClickable><CardHeader selectableActions={obj} /></Card>`,
```

Comment:
That's actually a good point. In this particular case the variable may not be shared with anything other than a CardHeader component since it's such a specific object (though wouldn't be surprised). But for something like a prop that accepts a color, maybe that's a valid case where a variable is being used in other places (maybe something like a "color" variable that possibly shouldn't be updated if it's being used in other places).

If anything we can base it on a case-by-case basis, but worth keeping in mind when trying to apply a fix to a variable like this.

Reply:
cc @wise-king-sullyman in case you had any thoughts